src/AfterLogin/StudyPage.js

```diff
@@ -0,0 +1,282 @@
+            <div className="study-card-header">
+              {/* <p>Spanish</p> */}
+              <img
+                src="/Spain.png"
+                alt="spanish-flag"
+                className="study-flag-icon "
+              />
+              <LoadingButton
+                loading={loadingAudio}
+                className="study-audio-loading"
+                onClick={() => {
+                  handlePlayAudio(currentCard.spanish);
+                }}
+              >
+                <VolumeUpIcon fontSize="large" />
+              </LoadingButton>
+            </div>
+            <div className="study-word" onClick={handleClick}>
+              <h1>{currentCard.spanish}</h1>
+            </div>
+          </Card>
+          <div className="study-container-button">
+            <Button
+              fullWidth
+              color="error"
+              size="large"
+              variant="contained"
+              onClick={handleRepeat}
+            >
+              👎Again
+            </Button>
+            <Button
+              color="primary"
+              fullWidth
+              size="large"
+              variant="contained"
+              onClick={handleNextCard}
+            >
+              {currentIndex === totalCards - 1 ? "Done" : "👍Good"}
+            </Button>
+          </div>
+        </>
+      )}
+    </>
+  );
+
+  const progressBar = ({ current, total }) => {
+    const progress = (current / total) * 100;
+    return (
+      <LinearProgress
+        variant="determinate"
+        value={progress}
+        color="error"
+        sx={{ marginBottom: 2 }}
+      />
+    );
+  };
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <ErrorPage
+          errorMessage={errorMessage}
+          handleErrorMessage={handleErrorMessage}
+        />
+        <Backdrop open={!deck.deckCards}>
+          <h3>Generating deck</h3>
+          <h1>
+            <CircularProgress color="inherit" />
+          </h1>
+        </Backdrop>
+
+        {deck.deckCards && Object.keys(deck.deckCards).length > 0 && (
+          <>
+            <div className="study-header">
+              <h2>{deck.deckName}</h2>
+              <FormGroup>
+                <FormControlLabel
+                  control={
+                    <Switch checked={autoPlay} onChange={handleChange} />
+                  }
+                  label="Auto-Play"
+                />
+              </FormGroup>
+            </div>
+
+            <p className="study-current-index">
+              {currentIndex + 1}/{totalCards}
+            </p>
+            {progressBar({ current: currentIndex + 1, total: totalCards })}
+
+            <div className="study-card">
+              {displayEnglish ? cardEnglish : cardSpanish}
+            </div>
+          </>
+        )}
+
+        {studyDone && (
+          <StudyDone open={studyDone} onClose={handleCloseStudyDone} />
+        )}
+      </div>
+    </ThemeProvider>
+  );
+}
```

Comment:
You have a few components with about 300 lines of code. This is too long I feel unless there is really a lot of complexity involved. I see lots of repetition that could be abstracted away to make things more readable. Try to think if you can refactor your code more while working on it. Give sections specific responsibilities, and abstract this into functions and components to make things more readable. If you would ask me to extend or edit a 300 line component, it would be very hard to do so without previous knowledge.